gitgud/operations.py

```diff
@@ -498,6 +509,15 @@ def get_copy_mapping(self, non_merges, known_commits):
 
         return mapping
 
+    def get_all_commits(self):
```

Comment:
Take a look at the code for `get_current_tree`. It seems like we can delete that first while loop and replace the second while loop with `for commit in file_operator.get_all_commits:`